yt_dlp/extractor/ant1newsgr.py

```diff
@@ -0,0 +1,164 @@
+        self._sort_formats(formats)
+        thumbnails = self._scale_thumbnails_to_max_width(
+            formats, [{'url': info['thumb']}], r'(?<=/imgHandler/)\d+')
+        return {
+            'id': video_id,
+            'title': info['title'],
+            'thumbnails': thumbnails,
+            'formats': formats,
+            'subtitles': subs,
+        }
+
+
+class Ant1NewsGrWatchIE(Ant1NewsGrBaseIE):
+    IE_NAME = 'ant1newsgr:watch'
+    IE_DESC = 'ant1news.gr videos'
+    _VALID_URL = r'https?://(?P<netloc>(?:www\.)?ant1news\.gr)/watch/(?P<id>\d+)/'
+    _API_PATH = '/templates/data/player'
+
+    _TEST = {
+        'url': 'https://www.ant1news.gr/watch/1506168/ant1-news-09112021-stis-18-45',
+        'md5': '95925e6b32106754235f2417e0d2dfab',
+        'info_dict': {
+            'id': '1506168',
+            'ext': 'mp4',
+            'title': 'md5:0ad00fa66ecf8aa233d26ab0dba7514a',
+            'description': 'md5:18665af715a6dcfeac1d6153a44f16b0',
+            'thumbnail': 'https://ant1media.azureedge.net/imgHandler/640/26d46bf6-8158-4f02-b197-7096c714b2de.jpg',
+        },
+    }
+
+    def _real_extract(self, url):
+        video_id, netloc = self._match_valid_url(url).group('id', 'netloc')
+        webpage = self._download_webpage(url, video_id)
+        info = self._download_and_extract_api_data(video_id, netloc)
+        info['description'] = self._og_search_description(webpage)
+        return info
+
+
+class Ant1NewsGrArticleIE(Ant1NewsGrBaseIE):
+    IE_NAME = 'ant1newsgr:article'
+    IE_DESC = 'ant1news.gr articles'
+    _VALID_URL = r'https?://(?:www\.)?ant1news\.gr/[^/]+/article/(?P<id>\d+)/'
+
+    _TESTS = [{
+        'url': 'https://www.ant1news.gr/afieromata/article/549468/o-tzeims-mpont-sta-meteora-oi-apeiles-kai-o-xesikomos-ton-kalogeron',
+        'md5': '294f18331bb516539d72d85a82887dcc',
+        'info_dict': {
+            'id': '_xvg/m_cmbatw=',
+            'ext': 'mp4',
+            'title': 'md5:a93e8ecf2e4073bfdffcb38f59945411',
+            'timestamp': 1603092840,
+            'upload_date': '20201019',
+            'thumbnail': 'https://ant1media.azureedge.net/imgHandler/640/756206d2-d640-40e2-b201-3555abdfc0db.jpg',
+        },
+    }, {
+        'url': 'https://ant1news.gr/Society/article/620286/symmoria-anilikon-dikigoros-thymaton-ithelan-na-toys-apoteleiosoyn',
+        'info_dict': {
+            'id': '620286',
+            'title': 'md5:91fe569e952e4d146485740ae927662b',
+        },
+        'playlist_mincount': 2,
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        info = self._search_json_ld(webpage, video_id, expected_type='NewsArticle')
+        embed_urls = list(Ant1NewsGrEmbedIE._extract_urls(webpage))
+        if not embed_urls:
+            raise ExtractorError('no videos found for %s' % video_id)
```

Comment:
In principle a news article page may or may not embed video(s) in this way. Should an error be raised in the latter case (like so), since the extractor has already matched the URL? Or should it return the URL and defer to other extractors?
The reason for this extractor over the generic extractor is the extraction of JSON-LD metadata.